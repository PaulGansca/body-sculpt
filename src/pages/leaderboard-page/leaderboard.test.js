import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import LeaderboardPage from './leaderboard-page';
import LeaderboardPageContainer from './leaderboard-page-container';

let wrapper;

beforeEach(() => {
    const mockProps = {
        users: []
    }
    wrapper = shallow(<LeaderboardPage {...mockProps} />);
})

describe('<LeaderboardPage />', () => {
    it("expect to render leaderboard page with props", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it("expect to render container with no props", () => {
        const mockProps = {
            getState: jest.fn(),
            subscribe: jest.fn(),
            dispatch: jest.fn(),
        }
        wrapper = shallow(
                    <Provider store={mockProps}>
                        <LeaderboardPageContainer {...mockProps} />
                    </Provider>)
        expect(wrapper).toMatchSnapshot();
    })
});