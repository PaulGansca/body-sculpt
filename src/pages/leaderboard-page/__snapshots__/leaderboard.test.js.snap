// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeaderboardPage /> expect to render container with no props 1`] = `ShallowWrapper {}`;

exports[`<LeaderboardPage /> expect to render leaderboard page with props 1`] = `ShallowWrapper {}`;
